File: smonitor-services/src/main/java/org/lorislab/smonitor/rs/model/ServerInfo.java
/*
 * Copyright 2013 lorislab.org.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.lorislab.smonitor.rs.model;

import java.util.List;

/**
 *
 * @author Andrej Petras
 */
public class ServerInfo {
    
    private String guid;
    
    /**
     * The id.
     */
    private String id;
    /**
     * The name.
     */
    private String name;
    
    /**
     * The server applications.
     */
    private List<ServerApplication> applications;

    public String getGuid() {
        return guid;
    }

    public void setGuid(String guid) {
        this.guid = guid;
    }
    
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<ServerApplication> getApplications() {
        return applications;
    }

    public void setApplications(List<ServerApplication> applications) {
        this.applications = applications;
    }
        
}

File: smonitor-web/src/main/webapp/assets/js/org/lorislab/smonitor/model/ServerInfo.js
/**
 * The server info model.
 * 
 * @constructor
 * @returns {ServerApplication} the new created server info.
 * @param {org.lorislab.smonitor.rs.model.ServerInfo} data the server side object server info.
 * 
 * @see org.lorislab.smonitor.rs.model.ServerInfo
 * 
 * @author Andrej Petras
 */
function ServerInfo(data) {
	this.guid = data.guid;
	this.id = data.id;
	this.name = data.name;
	this.applications = ko.observableArray();
	this.updateApp(data);
}

/**
 * Updates the list of applications from server.
 * @param {org.lorislab.smonitor.rs.model.ServerInfo} data
 */
ServerInfo.prototype.updateApp = function(data) {
	var tmp = $.map(data.applications, function(item) {
		return new ServerApplication(item);
	});
	this.applications(tmp);
};